manufacturing-ai-suite/industrial-edge-insights-time-series/docs/user-guide/how-to-securely-connect-to-on-premise-mqtt.md

```diff
@@ -0,0 +1,195 @@
+
+```
+"alerts": {
+        "mqtt": {
+            "mqtt_broker_host": "<mqtt broker host>",
+            "mqtt_broker_port": <mqtt broker port>,
+            "name": "my_mqtt_broker"
+        }
+    }
+```
+
+6. Volume mount the `kapacitor_devmode.conf`
+
+Volume mount the `edge-ai-libraries/microservices/time-series-analytics/config/kapacitor_devmode.conf` file in the `ia-time-series-analytics-microservice` container.
+Make the below changes to the `edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-time-series/docker-compose.yml` file.
+
+```yaml
+ia-time-series-analytics-microservice:
+  volumes:
+  - <absolute path to kapacitor_devmode.conf>:/app/config/kapacitor_devmode.conf
+```
+
+7. Comment the `ia-mqtt-broker` from the `depends_on` key in the `edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-time-series/docker-compose.yml` file.
+Eg is shown for the `ia-time-series-analytics-microservice` service.
+
+```yaml
+ia-time-series-analytics-microservice:
+  depends_on:
+      - ia-influxdb
+      # - ia-mqtt-broker
+```
+
+8. Comment the `ia-mqtt-broker` service in the edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-time-series/docker-compose.yml` file as shown below.
+
+```yaml
+# ia-mqtt-broker:
+  #   user: "${TIMESERIES_UID}:${TIMESERIES_UID}"
+  #   container_name: ia-mqtt-broker
+  #   hostname: ia-mqtt-broker
+  #   read_only: true
+  #   image: eclipse-mosquitto:2.0.21
+  #   restart: unless-stopped
+  #   depends_on:
+  #     - ia-influxdb
+  #   security_opt:
+  #   - no-new-privileges
+  #   healthcheck:
+  #     test: ["CMD-SHELL", "exit", "0"]
+  #     interval: 5m
+  #   networks:
+  #   - timeseries_network
+  #   volumes:
+  #   - ./configs/mqtt-broker/mosquitto.conf:/mosquitto/config/mosquitto.conf:ro
+  #   - ./certs:/run/secrets:ro
+
+```
+
+10. Replace `ia-mqtt-broker` in the `edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-time-series/docker-compose.yml` file with the external mqtt broker host ip.
+Example is as shown below.
+
+```yaml
+ia-telegraf:
+  environment:
+      no_proxy: "ia-influxdb,192.168.1.100,ia-opcua-server,ia-time-series-analytics-microservice,${no_proxy}"
+      NO_PROXY: "ia-influxdb,192.168.1.100,ia-opcua-server,ia-time-series-analytics-microservice,${no_proxy}"
+```
+
+11. Add environment variable `PORT` to the `ia-mqtt-publisher` service as shown below in the `edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-time-series/docker-compose.yml` file.
+Assign value of the external mqtt broker port to this variable.
+
+```yaml
+ia-mqtt-publisher:
+  environment:
+    AppName: "mqtt-publisher"
+    PORT: 8000
+```
+
+12. Update the external mqtt broker port in the nginx service in `edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-time-series/docker-compose.yml` file.
+Below is an example shown with port number 8000.
+
+```yaml
+nginx:
+  ports:
+  - 8000:8000
+```
+
+13. Update the `nginx.conf` file.
```

Comment:
This is not required, we don't want to enforce the 3rd party broker launched by them through our reverse proxy